lib/codegen_types.re

```diff
@@ -1,4 +1,64 @@
-open Common;
+module B = Ast_grammar_normalized;
 
-let codegen = _normalized_ast => 
-   raise(Todo);
+let counter = ref(0);
+let gen_intermediate_type = () => {
```

Comment:
This should be called gen_intermediate_constructor (that's the name of those A | B | C | ... in ocaml.
A type is 'expression', a constructor is Addition or Var or Number etc.

Reply:
will rename it